src/windows/wslasdk/wsla.h

```diff
@@ -0,0 +1,223 @@
+    PVOID progressCallbackContext;
+} WLSA_IMPORT_CONTAINER_IMAGE_OPTIONS;
+
+STDAPI WslaImportContainerImage(_In_ WslaSession session, _In_ const WLSA_PULL_CONTAINER_IMAGE_OPTIONS* options);
+
+typedef struct WSLA_CONTAINER_IMAGE_INFO
+{
+    PCSTR repository;
+    PCSTR tag;
+    UINT8 sha256[32];
+} WSLA_CONTAINER_IMAGE_INFO;
+
+STDAPI WslaListContainerImages(_In_ WslaSession sesssion, _Inout_ WSLA_CONTAINER_IMAGE_INFO* images, _Inout_ UINT32* count);
+
+STDAPI WslaDeleteContainerImage(_In_ WslaSession session, _In_ PCSTR imageName);
+
+// Container
+
+typedef struct WSLA_CONTAINER_PORT_MAPPING
+{
+    UINT16 windowsPort;
+    UINT16 containerPort;
+
+    // TODO: Port mapping type? Host/Bridge, etc
+} WSLA_CONTAINER_PORT_MAPPING;
+
+typedef struct WSLA_CONTAINER_VOLUME
+{
+    PCWSTR windowsPath;
+    PCSTR containerPath;
+} WSLA_CONTAINER_VOLUME;
+
+typedef struct WSLA_CONTAINER_GPU_OPTIONS
+{
+    BOOL enable;
+    PCSTR gpuDevices;
+} WSLA_CONTAINER_GPU_OPTIONS;
+
+typedef struct WSLA_CONTAINER_PROCESS_OPTIONS
+{
+    PCSTR executable; // Full path to executable inside container
+    PCSTR* commandLine;
+    UINT32 commandLineCount;
+    PCSTR* environment;
+    UINT32 environmentCount;
+    PCSTR currentDirectory;
+} WSLA_CONTAINER_PROCESS_OPTIONS;
+
+typedef struct WSLA_CONTAINER_OPTIONS
+{
+    PCSTR image; // Image name (repository:tag)
+    PCSTR name;  // Container runtime name (expected to allow DNS resolution between containers)
+    const WSLA_CONTAINER_PORT_MAPPING* ports;
+    UINT32 portsCount;
+    const WSLA_CONTAINER_VOLUME* volumes;
+    UINT32 volumesCount;
+    const WSLA_CONTAINER_GPU_OPTIONS* gpuOptions;
+    const WSLA_CONTAINER_PROCESS_OPTIONS* initProcessOptions;
+} WSLA_CONTAINER_OPTIONS;
+
+typedef struct WSLA_CONTAINER_PROCESS
+{
+    UINT32 pid;
+    HANDLE exitEvent;
+    HANDLE stdIn;
+    HANDLE stdOut;
+    HANDLE stdErr;
+} WSLA_CONTAINER_PROCESS;
+
+DECLARE_HANDLE(WslaRuntimeContainer);
+
+STDAPI WslaCreateNewContainer(
+    _In_ WslaSession session, _In_ const WSLA_CONTAINER_OPTIONS* options, _Out_ WslaRuntimeContainer* container, _Out_ WSLA_CONTAINER_PROCESS* initProcess);
+
+STDAPI WslaStartContainer(_In_ WslaRuntimeContainer container);
+
+STDAPI WslaStopContainer(_In_ WslaRuntimeContainer container);
+
+STDAPI WslaDeleteContainer(_In_ WslaRuntimeContainer container);
+
+STDAPI WslaRestartContainer(_In_ WslaRuntimeContainer container);
+
+typedef enum WSLA_CONTAINER_STATE
+{
+    WSLA_CONTAINER_STATE_INVALID = 0,
+    WSLA_CONTAINER_STATE_CREATED = 1,
+    WSLA_CONTAINER_STATE_RUNNING = 2,
+    WSLA_CONTAINER_STATE_EXITED = 3,
+    WSLA_CONTAINER_STATE_FAILED = 4,
+} WSLA_CONTAINER_STATE;
+
+STDAPI WslaGetContainerState(_In_ WslaRuntimeContainer container, _Out_ WSLA_CONTAINER_STATE* state);
+
+// Container Process
+
+STDAPI WslaCreateContainerProcess(_In_ WslaRuntimeContainer container, _In_ const WSLA_CONTAINER_PROCESS_OPTIONS* options, _Out_ WSLA_CONTAINER_PROCESS* process);

```

Comment:
We might want to follow the COM logic a bit closer to make the integration easier. 

Something like: 

```
WslaGetContainerInitProcess(WslaRuntimeContainer container, WSLA_CONTAINER_PROCESS* process);
```

might be easier to implement

Reply:
This one was the "nerdctl exec". I think you mean splitting CreateNewContainer into 2 methods? That works for me too.